bin/citrea/tests/e2e/mod.rs

```diff
@@ -574,3 +574,129 @@ async fn execute_blocks(
+
+    sequencer_client.send_publish_batch_request().await;
+    wait_for_l2_block(&sequencer_client, 1, None).await;
+
+    let (contract_address, contract, runtime_code) = {
+        let contract = SimpleStorageContract::default();
+        let runtime_code = sequencer_client
+            .deploy_contract_call(contract.byte_code(), None)
+            .await
+            .unwrap();
+        let deploy_contract_req = sequencer_client
+            .deploy_contract(contract.byte_code(), None)
+            .await
+            .unwrap();
+        sequencer_client.send_publish_batch_request().await;
+
+        let contract_address = deploy_contract_req
+            .get_receipt()
+            .await
+            .unwrap()
+            .contract_address
+            .unwrap();
+
+        (contract_address, contract, runtime_code)
+    };
+
+    {
+        let set_value_req = sequencer_client
+            .contract_transaction(contract_address, contract.set_call_data(42), None)
+            .await;
+        sequencer_client.send_publish_batch_request().await;
+        set_value_req.watch().await.unwrap();
+    }
+
+    let code = sequencer_client
+        .eth_get_code(contract_address, None)
+        .await
+        .unwrap();
+
+    assert_eq!(code.to_vec()[..runtime_code.len()], runtime_code.to_vec());
+
+    // reach the block at which the fork will be activated
+    for _ in 3..=100 {
+        sequencer_client.send_publish_batch_request().await;
+    }
+
+    // This should access the `code` and copy code over to `offchain_code` in EVM
+    let code = sequencer_client
+        .eth_get_code(contract_address, None)
+        .await
+        .unwrap();
+    assert_eq!(code.to_vec()[..runtime_code.len()], runtime_code.to_vec());
+
+    // Execute transaction on the contract living in `offchain_code`
+    {
+        let set_value_req = sequencer_client
+            .contract_transaction(contract_address, contract.set_call_data(50), None)
+            .await;
+        sequencer_client.send_publish_batch_request().await;
+        set_value_req.watch().await.unwrap();
+    }
+
+    let code = sequencer_client
+        .eth_get_code(contract_address, None)
+        .await
+        .unwrap();
+    assert_eq!(code.to_vec()[..runtime_code.len()], runtime_code.to_vec());
+
+    // Deploy a contract post-fork
+    let (contract_address, contract) = {
+        let contract = SimpleStorageContract::default();
+        let deploy_contract_req = sequencer_client
+            .deploy_contract(contract.byte_code(), None)
+            .await
+            .unwrap();
+        sequencer_client.send_publish_batch_request().await;
+
+        let contract_address = deploy_contract_req
+            .get_receipt()
+            .await
+            .unwrap()
+            .contract_address
+            .unwrap();
+
+        (contract_address, contract)
+    };
+
+    {
+        let set_value_req = sequencer_client
```

Comment:
we should get the value and see if the tx actually worked. if the code couldn't be reached for example, the tx would go through but value wont be set